### PR TITLE
Temporal: Add additional coverage based on issues reported by Anba

### PR DESCRIPTION
**Pull request by ptomato**
Best reviewed commit by commit.

cc @anba

### CODE REVIEW

File: test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const legacyDate = new Date(sampleEpochMs);
+const instant = new Temporal.Instant(sampleEpochNs);
+const plainDateTime = new Temporal.PlainDateTime(2024, 9, 19, 12, 23, 37, 847);
+const plainDate = new Temporal.PlainDate(2024, 9, 19);
+const plainYearMonth = new Temporal.PlainYearMonth(2024, 9);
+const plainMonthDay = new Temporal.PlainMonthDay(9, 19);
+const plainTime = new Temporal.PlainTime(12, 23, 37, 847);
+
+const dateStyleFormatter = new Intl.DateTimeFormat(undefined, { dateStyle: "short", calendar: "iso8601", timeZone: "America/Vancouver" });
+const dateStyleResult = JSON.stringify(dateStyleFormatter.formatRangeToParts(legacyDate, legacyDate));
```
Comment (Ms2ger) [inline test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js:23]: Is the serialization order guaranteed (or reliable enough?)

File: test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+assert.sameValue(JSON.stringify(dateStyleFormatter.formatRangeToParts(instant, instant)), dateStyleResult, "Instant with dateStyle");
+assert.sameValue(JSON.stringify(dateStyleFormatter.formatRangeToParts(plainDateTime, plainDateTime)), dateStyleResult, "PlainDateTime with dateStyle");
+assert.sameValue(JSON.stringify(dateStyleFormatter.formatRangeToParts(plainDate, plainDate)), dateStyleResult, "PlainDate with dateStyle");
+assert.notSameValue(JSON.stringify(dateStyleFormatter.formatRangeToParts(plainYearMonth, plainYearMonth)), dateStyleResult, "PlainYearMonth with dateStyle should not throw but day is omitted");
+assert.notSameValue(JSON.stringify(dateStyleFormatter.formatRangeToParts(plainMonthDay, plainMonthDay)), dateStyleResult, "PlainMonthDay with dateStyle should not throw but year is omitted");
+assert.throws(TypeError, () => JSON.stringify(dateStyleFormatter.formatRangeToParts(plainTime, plainTime)), "no overlap between dateStyle and PlainTime");
```
Comment (Ms2ger) [inline test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js:30]: The JSON.stringify call doesn't seem useful for error cases

File: test/intl402/Temporal/Instant/prototype/toLocaleString/default-includes-time-not-time-zone-name.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+description: Tests what information is present by default
+locale: [en]
+features: [Temporal]
+---*/
+
+const instant = new Temporal.Instant(1735213600_321_000_000n); // 2024-12-26T11:46:40.321Z
+const result = instant.toLocaleString("en", { timeZone: "UTC" });
+
+assert(result.includes("2024"), `Instant formatted with no options ${result} should include year`);
+assert(result.includes("12") || result.includes("Mar"), `Instant formatted with no options ${result} should include month`);
```
[the remark this answers is not in the document]
Comment (ptomato) [reply test/intl402/Temporal/Instant/prototype/toLocaleString/default-includes-time-not-time-zone-name.js:15]: oh right :facepalm: I originally had this date in March but changed it to be a 2-digit number so I didn't have to worry about the possible lack of leading 0.

File: test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js (hunk 1 above)
Comment (ptomato) [reply test/intl402/DateTimeFormat/prototype/formatRangeToParts/temporal-objects-not-overlapping-options.js:23]: It's "ascending chronological order of property creation" and the order in which `type`, `value`, and `source` are created is specified, so I think we're good.

**Review by Ms2ger — APPROVED** [review]
Thanks!

**Pull request merged**